src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1045,6 +1045,16 @@ private object DeepCopierHelper(Type t, object original, ICopyContext context)
                 ". Perhaps you need to mark it [Serializable] or define a custom serializer for it?");
         }
 
+        /// <summary>        
+        /// This method exposes internal IsOrleansShallowCopyable to allow external custom serializers to check if is it ok to shallow copy objects when deep copying.               
```

Comment:
`Returns true if the provided type can be safely shallow-copied; otherwise false, indicating that the type must instead be deep-copied.`